samcli/lib/providers/sam_api_provider.py

```diff
@@ -148,7 +148,14 @@ def _get_cors_prop(cors_dict, prop_name):
         """
         prop = cors_dict.get(prop_name)
         if prop:
-            if (not isinstance(prop, string_types)) or (not (prop.startswith("'") and prop.endswith("'"))):
+            if not isinstance(prop, string_types) or prop.startswith("!"):
```

Comment:
Some intrinsic functions starts with `Fn::`, should we check for those as well?

Reply:
That are covered under the first check. Basically, anything that isn't a string gets caught. For `FN::` intrinsics, they are all dictionaries (key is `FN::`, value is a string, list, or dictionary).